app/main.py

```diff
@@ -1,34 +1,83 @@
+        if not (start[0] == end[0] or start[1] == end[1]):
+            raise ValueError("Ships shouldn't be located "
+                             "in the neighboring cells "
+                             "(even if cells are neighbors "
+                             "by diagonal) and has length 1 to 4")
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        self.decks = {(i, j): Deck(i, j)
+                      for i in range(start[0], end[0] + 1)
+                      for j in range(start[1], end[1] + 1)}
+        self.is_drowned = is_drowned
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        return self.decks[(row, column)]
+
+    def fire(self, row: int, column: int) -> None:
+        self.get_deck(row, column).is_alive = False
+
+        if not any(deck.is_alive for deck in self.decks.values()):
+            self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list[tuple]) -> None:
+        self.ships = {(x, y): Ship(x, y) for x, y in ships}
+        self.field = {}
+        self.field.update({deck: ship
+                           for ship in self.ships.values()
+                           for deck in ship.decks.keys()})
+        self._validate_field()
+
+    def fire(self, location: tuple) -> str:
+
+        if not self.field.get(location):
+            return "Miss!"
+
+        self.field[location].fire(*location)
+
+        if self.field[location].is_drowned:
+            return "Sunk!"
+
+        return "Hit!"
+
+    def print_field(self) -> None:
+        for row in range(1, 11):
+            for column in range(1, 11):
+                point = (row, column)
+                if self.field.get(point) is None:
+                    print("~\t", end="")
+                    continue
+                if self.field[point].is_drowned:
+                    print("x\t", end="")
+                    continue
+                if not self.field[point].decks[point].is_alive:
+                    print("*\t", end="")
+                else:
+                    print(u"\u25A1" + "\t", end="")
+            print("\n")
+
+    def _validate_field(self) -> None:
+        # validate data is dict{decks: count of ship}
+        validate = {4: 1, 3: 2, 2: 3, 1: 4}
```

Comment:
Variable names are usually nouns, not verbs. I'd suggest to rename this one into something in the spirit `correct_ship_count`. The name provided is only for you to get the idea.